### PR TITLE
feat: Add output formatting, CLI options, and  improved view engine check

### PR DESCRIPTION
**Pull request by Jordan-Hall**
The latest update to the script has pivoted from using semantic versioning checks to directly verifying Angular compatibility based on the presence of View Engine-specific outputs. This shift ensures a 100% accuracy rate in determining whether a dependency can be upgraded in the context of Angular 16's move away from the View Engine, which is crucial given the removal of `ngcc` and the sole support for the Ivy compilation and rendering pipeline in Angular 16.

### Major Changes
- **Detection of View Engine Outputs**: The script now examines package distributions for View Engine-specific metadata, replacing the previous method of relying on semantic versioning. This direct approach eliminates the ambiguity in compatibility checks, providing a clear and accurate assessment of whether a dependency supports Ivy.
- **Enhanced Compatibility Analysis**: By focusing on the technical requirements for Ivy compatibility, the script offers an improved mechanism to audit Angular project dependencies. This ensures that projects can confidently upgrade to Angular 16, fully aware of the dependencies that support the Ivy engine.
- **Command-Line Arguments**: Added support for `--output`, `--style`, `--help` (`-h`), and `--version` (`-v`) arguments, allowing users to customize the script's behavior directly from the command line.
- **Help Message**: A detailed help message (`-h`, `--help`) now provides users with information on available commands and their usage.
- **Version Information**: The `-v` or `--version` argument displays the current script version, aligning with the project's semantic versioning.
- **Output Formatting**: Introduced colorized output for enhanced readability in console mode. Markdown formatting is now supported for file outputs, particularly useful when specifying the `--style=markdown` option.
- **Dynamic Table Formatting**: For `--style=table`, table column widths are dynamically calculated to accommodate the length of content, ensuring a clean and organized display.

### Implementation Highlights
- The **`isViewEngine`** function meticulously checks for the existence of `.metadata.json` files associated with TypeScript definition files in package distributions. This presence indicates a View Engine-specific artifact, guiding the determination of Ivy compatibility.
- **Output formatting enhancements**, including dynamic table formatting and markdown output support, are tailored to present the compatibility analysis in a clear, organized, and visually appealing manner. These features facilitate easier interpretation and sharing of the results.
- The **`colorize` function** and **`writeToOutput` utility** significantly boost the script's usability by offering flexible, attractive output presentation, whether in console logs or written to files for documentation purposes.

With these enhancements, the script stands as a vital tool for developers and teams planning to upgrade to Angular 16, providing precise insights into dependency compatibility with the Ivy engine. This focused approach assists in identifying and addressing potential upgrade barriers, supporting a smoother transition to the latest Angular version.

### CODE REVIEW

**Comment by Jordan-Hall** [comment]
Markdown preivew
image

I was thinking maybe add a link to the npm package 


Table format

image

File: index.js [hunk 1]
```diff
@@ -200,42 +248,126 @@ async function getDependenciesAndCheckCompatibility() {
...
+	}
+}
+
+function printMarkdownOutput(dependenciesToCheck) {
+	if (dependenciesToCheck.mayNeedUpgrade.length > 0) {
+		writeToOutput(`### Dependencies that are maintained but may need upgrading`);
+		writeToOutput(`| Package | Current Version | Latest Version |`);
+		writeToOutput(`| ------- | --------------- | -------------- |`);
+		dependenciesToCheck.mayNeedUpgrade.forEach(dep => {
+			writeToOutput(`| ${dep.packageName} | ${dep.currentVersion} | ${dep.latestVersion} |`);
```
Comment (Jordan-Hall) [inline index.js]:
```suggestion
			writeToOutput(`| [${dep.packageName}](https://npmjs.com/package/${dep.packageName}) | ${dep.currentVersion} | ${dep.latestVersion} |`);
```

File: index.js [hunk 2]
```diff
@@ -200,42 +248,126 @@ async function getDependenciesAndCheckCompatibility() {
...
+		});
+		writeToOutput('\n');
+	}
+
+	if (dependenciesToCheck.reviewForRemoval.length > 0) {
+		writeToOutput(`### Dependencies to review for removal or replacement`);
+		writeToOutput(`| Package | Current Version | Latest Version |`);
+		writeToOutput(`| ------- | --------------- | -------------- |`);
+		dependenciesToCheck.reviewForRemoval.forEach(dep => {
+			writeToOutput(`| ${dep.packageName} | ${dep.currentVersion} | ${dep.latestVersion} |`);
```
Comment (Jordan-Hall) [inline index.js]:
```suggestion
			writeToOutput(`| [${dep.packageName}](https://npmjs.com/package/${dep.packageName}) | ${dep.currentVersion} | ${dep.latestVersion} |`);
```

File: index.js [hunk 3]
```diff
@@ -200,42 +248,126 @@ async function getDependenciesAndCheckCompatibility() {
...
+		});
+		writeToOutput('\n');
+	}
+
+	if (dependenciesToCheck.unknown.length > 0) {
+		writeToOutput(`### Dependencies without @angular/core or dependencies visible in NPM registry`);
+		dependenciesToCheck.unknown.forEach(dep => {
+			writeToOutput(`- ${dep}`);
```
Comment (Jordan-Hall) [inline index.js]:
```suggestion
			writeToOutput(`- [${dep}](https://npmjs.com/package/${dep})`);
```

**Review by Jordan-Hall — COMMENTED** [review]
@danielglejzner Maybe good to be able to click through and go to the npm package?

**Review by danielglejzner — APPROVED** [review]

**Pull request merged**